mozilla/fx-private-relay
# MPP-3722, MPP-3513, MPP-3890, MPP-3373: Handle more errors when relaying SMS messages

This PR addresses a few tickets, to remove some 500 errors from the service:

* MPP-3722: Changes the `RelaySMSException` classes to be based on `RelayAPIException`
  - Also sets `ftl_id` on the exception classes, to make it easier to cross-reference exceptions to the translated versions
  - Also move exceptions to `/phones/exceptions.py`
  - Sync `NoPhoneLog` message with translations
  - Update `TemplateTwiMLRenderer` to handle `RelaySMSException` without having to wrap in `rest_framework.exceptions.ValidationError`
  - Skip title case for `RelaySMSException`, keep for `ValidationError`
* MPP-3513: `ShortPrefixMatchesNoSenders` - Log at INFO instead of sending an exception to Sentry. This is done for all exceptions derived from `RelaySMSException`, for Twilio and iQ
* MPP-3890: When the number for a previous contact can not be parsed by the `phonenumbers`, skip it rather than raise the exception
* MPP-3373 (addressed but not fixed) - When a Relay user has sent `STOP` to their Relay number, we can't send them any more messages. Log this as INFO and reply with an empty "OK" response to Twilio. This prevents the 500 error, but doesn't help the Relay user discover the issue. An obvious next step is to set a flag on the `RelayNumber` or `RealNumber` that the user has blocked their Relay number, so we can show that in the frontend.
* Fix some typos detected by [typos](https://github.com/crate-ci/typos)

If this is too much for one PR, I can break it up.

How to test:

- [x] l10n changes have been submitted to the l10n repository, if any - https://github.com/mozilla-l10n/fx-private-relay-l10n/pull/174
- ~I've added or updated relevant docs in the docs/ directory.~
- ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
